### PR TITLE
execution plan bug fix

### PR DESCRIPTION
**Pull request by gituser173**
Signed-off-by: Claudiu Ban claudiu-laurian.ban@hpe.com

### CODE REVIEW

File: cloudslang-entities/src/main/java/io/cloudslang/lang/entities/bindings/values/SensitiveValue.java [hunk 1]
```diff
@@ -29,6 +31,7 @@
 public class SensitiveValue implements Value {
 
     public static final String SENSITIVE_VALUE_MASK = "********";
+    public static final String ENCRYPTOR_PACKAGE = "com.hp.onyx.commons.encryption";
```
Comment (Bonczidai) [inline cloudslang-entities/src/main/java/io/cloudslang/lang/entities/bindings/values/SensitiveValue.java]: I still don't think we should put company related info in open source and make CS dependent on OO codebase - I would rather just pass a single constructor param to set the mode -- encryption / obfuscation or maybe a System Property -- that way we could easily add unit tests at CS level -- if somebody wants to modify the obfuscation / enc behavior he can do it anyway

@lucian-cm WDYT?

File: cloudslang-entities/src/main/java/io/cloudslang/lang/entities/encryption/DummyEncryptor.java [hunk 2]
```diff
@@ -19,6 +19,11 @@ public String encrypt(char[] clearText) {
     }
 
     @Override
+    public String obfuscate(String cypherText) {
```
Comment (gituser173) [inline cloudslang-entities/src/main/java/io/cloudslang/lang/entities/encryption/DummyEncryptor.java]: no, the decrypt method does deObfuscation and decrypt

File: cloudslang-entities/src/main/java/io/cloudslang/lang/entities/bindings/values/SensitiveValue.java [hunk 3]
```diff
@@ -40,6 +43,8 @@
     @JsonIgnore
     private Serializable originalContent = null;
 
+    private boolean isExecutionPlan;
```
Comment (gituser173) [inline cloudslang-entities/src/main/java/io/cloudslang/lang/entities/bindings/values/SensitiveValue.java]: this value is false by default which is the case for all other constructors
@lucian-cm WDYT?

**Comment by lucian-cm** [comment]
@gituser173 , as a general comment, I fear that the method modified io.cloudslang.lang.compiler.modeller.transformers.AbstractInputsTransformer#createInput(java.lang.String, java.io.Serializable, boolean, boolean, boolean) is actually used from several points in the CloudSlang codebase.

This makes this pull request a bit unsafe and probably not equivalent.
I agree with @Bonczidai comment about the package.

What I think the best solution is for this case is to make an entry point method for the specific compile service from where you pass as an enum the type of data operation to be done (for encryption or obfuscation) and this ensures that CloudSlang code remains the same. And these removes the need to put that specific package reference.

File: cloudslang-entities/src/main/java/io/cloudslang/lang/entities/bindings/values/SensitiveValue.java (hunk 1 above)
Comment (lucian-cm) [inline cloudslang-entities/src/main/java/io/cloudslang/lang/entities/bindings/values/SensitiveValue.java]: @Bonczidai totally agree

File: cloudslang-entities/src/main/java/io/cloudslang/lang/entities/bindings/values/SensitiveValue.java (hunk 3 above)
Comment (lucian-cm) [inline cloudslang-entities/src/main/java/io/cloudslang/lang/entities/bindings/values/SensitiveValue.java]: I prefer constructors as well, but surely an = initializer will work as well.

**Pull request closed without merging**